app/services/wallets/validate_service.rb

```diff
@@ -78,7 +78,11 @@ def valid_metadata?
     end
 
     def valid_limitations?
-      return true if Wallets::ValidateLimitationsService.new(BaseService::Result.new, **args).valid?
+      limitation_result = BaseService::Result.new
+      limitation_result.billable_metrics = result.billable_metrics
+      limitation_result.billable_metric_identifiers = result.billable_metric_identifiers
+
+      return true if Wallets::ValidateLimitationsService.new(limitation_result, **args).valid?
```

Comment:
Is there a reason why we don't use `result` directly ?

```suggestion
      return true if Wallets::ValidateLimitationsService.new(result, **args).valid?
```